moveit_simple/include/moveit_simple/robot.h

```diff
@@ -433,6 +433,32 @@ class Robot
    */
   double getSpeedModifier(void) const;
 
+  /**
+   * @brief limitJointWindup - Enable/Disable joint windup reduction in IK solutions by
+   * applying the seed state modifiers defined in ik_seed_state_fractions_. The fraction values
+   * are multiplied with the corresponding joint values in the seed state which causes IK solutions
+   * to be "pulled" towards the center. Additionally the modified joints are solved with a lower
+   * consistency_limit of PI. Both measures in combination act as a "soft" joint limit which doesn't
+   * affect the actual freedom of the joint but only centers IK solutions by a specified amount.
+   * @param enabled - Set joint windup limitation feature on/off
+   */
+  void limitJointWindup(bool enabled);
+
+  /**
+   * @brief setIKSeedStateFractions - Specify joint value modifiers that should be applied in IK
+   * calls if joint windup limitation is enabled. On how to enable and use this feature see the function
+   * limitJointWindup() above.
+   * @param ik_seed_state_fractions - A map from joint ids to fraction values in range [0,1]
+   * @return true on success, false if invalid entries are found (invalid joint id, value out of range)
+   */
+  bool setIKSeedStateFractions(const std::map<size_t, double>& ik_seed_state_fractions);
```

Comment:
These parameters seem too low level for the intended API Shawn wants for the Robot class. Seems to me settings for this should be set in a yaml and just assumed correct from the Robot class.

Reply:
Sure, I thought the same thing. I will add a demo config file to the test cases.